app/Controllers/Http/AuthController.ts

```diff
@@ -0,0 +1,57 @@
+import UndefinedFieldError from "App/Exceptions/CustomErrors/UndefiendFieldError";
+import User from "App/Models/User";
+import UserRegisterValidator from "App/Validators/UserRegisterValidator";
+
+export default class AuthController {
+  //For Registering the user
+  public async register({ request, response }) {
+    try {
+      let { email, password } = await request.validate(UserRegisterValidator);
+      
+      //Handling only space contained field,which return undefined after validation
+      if (email === undefined || password === undefined) {
+        const fieldName = !email ? "email" : "password";
+        throw new UndefinedFieldError(`${fieldName} required`, fieldName);
```

Comment:
if `email` or `password` is undefined. The validator will throw errors. Why we need to throw the errors explicitly?